lucene/core/src/java/org/apache/lucene/search/comparators/DoubleComparator.java

```diff
@@ -61,8 +63,12 @@ public LeafFieldComparator getLeafComparator(LeafReaderContext context) throws I
   /** Leaf comparator for {@link DoubleComparator} that provides skipping functionality */
   public class DoubleLeafComparator extends NumericLeafComparator {
 
+    private final byte[] deltaOne;
+
     public DoubleLeafComparator(LeafReaderContext context) throws IOException {
       super(context);
+      deltaOne = new byte[Double.BYTES];
+      DoublePoint.encodeDimension(1d, deltaOne, 0);
```

Comment:
This doesn't look correct to me. E.g. if the bottom value is 4 and the sort is ascending, this would mistakenly ignore 3.5. We need to go to the next double, which can be done by subtracting 1 to the encoded representation:
```suggestion
      deltaOne[deltaOne.length-1] = 1;
```

Reply:
addressed in [ef9f917](https://github.com/apache/lucene/pull/12405/commits/ef9f9173bbba04871dd489187cbdc26505d64418)